extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/VertxRequestHandler.java

```diff
@@ -165,22 +207,23 @@ private void processCloudEvent(RoutingContext routingContext) {
                 final ObjectWriter writer = (ObjectWriter) invoker.getBindingContext().get(DATA_OBJECT_WRITER);
 
                 final CloudEvent<?> inputCloudEvent;
+                if (binaryCE) {
```

Comment:
Does this need to be pull out of the if?

Reply:
Before `inputCloudEvent` was initialized conditionally , if was created only if the function accepted some input, `invoker.hasInput() == true`, otherwise it was set to null in `else` branch.

Reply:
ah you're right, make sense to move it into the if, thanks for details